lib/nanoc/cli/commands/nanoc.rb

```diff
@@ -10,6 +10,10 @@
   Nanoc::CLI.debug = true
 end
 
+opt :e, :env, 'set environment', argument: :optional do |value|
```

Comment:
I don’t think it makes sense for `-e` to have an optional argument, since it defaults to `:default` anyway. Make the argument required, and remove the `|| :default` check below.